structopt-derive/src/attrs.rs

```diff
@@ -603,6 +642,20 @@ impl Attrs {
     }
 }
 
+fn fresh_id() -> usize {
+    use std::cell::Cell;
+
+    thread_local! {
+        static NEXT_ID: Cell<usize> = Cell::new(0);
+    }
+
+    NEXT_ID.with(|next_id| {
+        let id = next_id.get();
+        next_id.set(id + 1);
+        id
+    })
```

Comment:
Using an atomic would be simpler and safer if you imagine that there is several threads generating ids. But well, that's just not really important in this case.

Reply:
Good point, but wee don't need it anymore